src/features/tracing/traceDatabase.ts

```diff
@@ -126,9 +148,14 @@ const downloadRemote = async () => {
         );
     }
 
-    remoteDatabasesCache = config.firmwares.devices.flatMap(
-        device => device.versions
-    );
+    remoteDatabasesCache = config.firmwares.devices
+        .filter(
+            device =>
+                nrfDeviceVersion === undefined ||
+                device.type === nrfDeviceVersion
+        )
+        .flatMap(device => device.versions);
```

Comment:
Seems the logic between remote and local cache is somewhat reused, could they be DRY'd up?